causalml/feature_selection/filters.py

```diff
@@ -0,0 +1,370 @@
+                )
+        return d_res
+
+
+    def _filter_D_one_feature(self, data, X_name, y_name, 
+                              n_bins=10, method='KL', control_group='control',
+                              experiment_group_column='treatment_group_key'):
+        """
+        Calculate the chosen divergence measure for one feature.
+        """
+        # [TODO] Application to categorical features
+
+        if method == 'KL':
+            evaluationFunction = self._evaluate_KL
+        elif method == 'ED':
+            evaluationFunction = self._evaluate_ED
+        elif method == 'Chi':
+            evaluationFunction = self._evaluate_Chi
+
+        totalSize = len(data.index)
+        x_bin = pd.qcut(data[X_name].values, n_bins, labels=False, 
+                        duplicates='drop')
+        d_children = 0
+        for i_bin in range(x_bin.max() + 1): # range(n_bins):
+            nodeSummary = self._GetNodeSummary(
+                data=data.loc[x_bin == i_bin], 
+                experiment_group_column=experiment_group_column, y_name=y_name
+            )[1]
+            nodeScore = evaluationFunction(nodeSummary, 
+                                           control_group=control_group)
+            nodeSize = sum([x[1] for x in list(nodeSummary.values())])
+            d_children += nodeScore * nodeSize / totalSize
+
+        parentNodeSummary = self._GetNodeSummary(
+            data=data, experiment_group_column=experiment_group_column, y_name=y_name
+        )[1]
+        d_parent = evaluationFunction(parentNodeSummary, 
+                                      control_group=control_group)
+            
+        d_res = d_children - d_parent
+        
+        D_result = pd.DataFrame({
+            'feature': X_name, 
+            'method': method,
+            'score': d_res, 
+            'p_value': None,
+            'misc': 'number_of_bins: {}'.format(min(n_bins, x_bin.max()+1)),# format(n_bins),
+        }, index=[0]).reset_index(drop=True)
+
+        return(D_result)
+
+    def filter_D(self, data, X_name_list, y_name, 
+                 n_bins=10, method='KL', control_group='control',
+                 experiment_group_column='treatment_group_key'):
+        """
+        Rank features based on the chosen divergence measure.
+        """
+        
+        all_result = pd.DataFrame()
+
+        for x_name_i in X_name_list: 
+            one_result = self._filter_D_one_feature(
+                data=data, X_name=x_name_i, y_name=y_name,
+                n_bins=n_bins, method=method, control_group=control_group,
+                experiment_group_column=experiment_group_column, 
+            )
+            all_result = pd.concat([all_result, one_result])
+
+        all_result = all_result.sort_values(by='score', ascending=False)
+        all_result['rank'] = all_result['score'].rank(ascending=False)
+
+        return all_result
+
+    def get_importance(self, data, X_name_list, y_name, method, 
```

Comment:
More generally, I think it would make sense to use same conventions as elsewhere in the package, ie use "X", "treatment", "y" and "control_name" as the argument names.

Reply:
hmm, one difference is that, in other functions, X and y are values in list format, but X_name_list and y_name is string here. The consideration is that feature importance will come with feature name as outputs for interpretation, therefore only values will be insufficient. Instead, we require inputs in (data, X_name_list, y_name) format.

Reply:
True. In terms of pre-existing convention, it looks like we're using 'features' for a list of feature names in the explainer class here: https://github.com/uber/causalml/blob/master/causalml/inference/meta/explainer.py For y_name, I'm not sure that any convention exists.